### PR TITLE
Content editable

### PR DESCRIPTION
**Pull request by yerramshilpa**
## Type of Change (Select One and Apply Github Label)

- [ ] Bug fix (non-breaking change which fixes an issue) -> Apply bugfix label
- [ ] New feature (non-breaking change which adds functionality) -> Apply enhancement label
- [ ] Breaking change (fix or feature that would cause existing functionality to change) -> Apply breaking label

## Which browsers have you tested?
- [ ] Internet Explorer 11
- [ ] Edge
- [ ] Chrome
- [ ] Firefox
- [ ] Safari

## Checklist:

- [ ] I have read the (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md] document.
- [ ] My code passes the automated linter.
- [ ] This code has been reviewed by another team member and passes the reviewer checklist found in (https://github.comv/GSA/sam-ui-elements/blob/CONTRIBUTING.md)[CONTRIBUTING.md]
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code is 508 compliant as tested by AMP and JAWS
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/packages/components/src/lib/editor/editor.component.ts [hunk 1]
```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
...
+
+    // onKeydown(event) {
+    //     if (event.key === 'Backspace' || event.key === 'Delete') {
+    //         this.length = this.length - 1;
+    //     }
+    // }
+
+    // onKeyPressed() {
+    //     this.length = this.length + 1;
+    // }
```
Comment (Bibek-Regmi) [inline libs/packages/components/src/lib/editor/editor.component.ts:130]: Let's get rid of commented out code

File: libs/packages/components/src/lib/editor/editor.component.ts [hunk 2]
```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
...
+        let caretOffset = 0;
+        if (typeof window.getSelection != 'undefined') {
+            let range = window.getSelection().getRangeAt(0);
+            let preCaretRange = range.cloneRange();
+            preCaretRange.selectNodeContents(element);
+            preCaretRange.setEnd(range.endContainer, range.endOffset);
+            caretOffset = preCaretRange.toString().length;
+        } else if (
+            typeof this._document['selection'] != 'undefined' &&
+            this._document['selection'].type != 'Control'
```
Comment (Bibek-Regmi) [inline libs/packages/components/src/lib/editor/editor.component.ts:143]: Can you add comment for this - in what situation would `window.getSelection` be undefined vs `document.selection`?

File: libs/packages/components/src/lib/editor/editor.component.ts [hunk 3]
```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
...
+            let childNodeIndex;
+            if (this.pos < firstNodeLength) {
+                childNodeIndex = 0;
+            } else if (this.pos > firstNodeLength) {
+                childNodeIndex = 2;
+            }
+            let startPosition =
+                childNodeIndex === 0 ? this.pos : this.pos - firstNodeLength - 1;
+            let range = this._document.createRange();
+            let sel = window.getSelection();
```
Comment (Bibek-Regmi) [inline libs/packages/components/src/lib/editor/editor.component.ts:196]: Are we assuming `window.getSelection()` is always defined here?

File: libs/packages/components/src/lib/editor/editor.component.ts [hunk 4]
```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
...
+            preCaretTextRange.moveToElementText(element);
+            preCaretTextRange.setEndPoint('EndToEnd', textRange);
+            caretOffset = preCaretTextRange.text.length;
+        }
+        return caretOffset;
+    }
+
+    onValueChange(value) {
+
+        const rawValue = value
```
Comment (Bibek-Regmi) [inline libs/packages/components/src/lib/editor/editor.component.ts:157]: Can we move some of the logic here to a separate function and have `validateRegex` and this method share that function? It seems like this logic of doing replaceAll, and then regex check, and then inserting `mark` element is repeated in both methods

File: libs/packages/components/src/lib/editor/editor.component.ts [hunk 5]
```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
...
+                rawValue.substring(index, index + 1) +
+                '</mark>' +
+                rawValue.substring(index + 1, index + rawValue.length);
+        } else {
+            res = rawValue;
+        }
+        this.searchInput.nativeElement.innerHTML = res;
+        this.model = rawValue;
+        this.updateModel();
+        let node = this._document.getElementById(this.id);
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/editor/editor.component.ts]: Updated in all places

File: libs/packages/components/src/lib/editor/editor.component.ts (hunk 2 above)
Comment (yerramshilpa) [reply libs/packages/components/src/lib/editor/editor.component.ts]: This condition never happens, removed the code

File: libs/packages/components/src/lib/editor/editor.component.ts [hunk 6]
```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
...
+        this.length = node.innerText.length;
+        if (this.pos < this.length) {
+            let firstNodeLength = node.childNodes[0].textContent.length;
+            let childNodeIndex;
+            if (this.pos < firstNodeLength) {
+                childNodeIndex = 0;
+            } else if (this.pos > firstNodeLength) {
+                childNodeIndex = 2;
+            }
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/editor/editor.component.ts:158]: Updated the condition to handle the equal length

File: libs/packages/components/src/lib/editor/editor.component.ts [hunk 7]
```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
...
+            let startPosition =
+                childNodeIndex === 0 ? this.pos : this.pos - firstNodeLength - 1;
+            let range = this._document.createRange();
+            let sel = window.getSelection();
+            range.setStart(
+                this._document.getElementById(this.id).childNodes[childNodeIndex],
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/editor/editor.component.ts]: refactored

File: libs/packages/components/src/lib/editor/editor.component.ts [hunk 8]
```diff
@@ -117,4 +118,94 @@ export class SdsEditorComponent implements ControlValueAccessor {
...
+                startPosition
+            );
+            range.collapse(true);
+            sel.removeAllRanges();
+            sel.addRange(range);
+            this._document.getElementById(this.id).focus();
```
[the remark this answers is not in the document]
Comment (yerramshilpa) [reply libs/packages/components/src/lib/editor/editor.component.ts]: Replaced in all the places

**Pull request merged**